### PR TITLE
plugins/plantuml-syntax: migrate to mkVimPlugin

### PR DESCRIPTION
**Pull request by GaetanLepage**

### CODE REVIEW

**Review by khaneliman — APPROVED** [review]

**Comment by GaetanLepage** [comment]
@mergify queue

File: plugins/by-name/plantuml-syntax/default.nix [hunk 1]
```diff
@@ -1,44 +1,44 @@
...
     };
   };
 
-  config =
-    let
-      cfg = config.plugins.plantuml-syntax;
-    in
-    mkIf cfg.enable {
-      extraPlugins = [ cfg.package ];
+  extraConfig = cfg: { extraPackages = [ cfg.plantumlPackage ]; };
```
Comment (khaneliman) [inline plugins/by-name/plantuml-syntax/default.nix:28]: I forgot, though... does this need a null check ?

Comment (GaetanLepage) [reply plugins/by-name/plantuml-syntax/default.nix:28]: Nope, our `extraPackages` option handles `null` values.

File: plugins/by-name/plantuml-syntax/default.nix [hunk 2]
```diff
@@ -1,44 +1,44 @@
...
 
-      globals = {
-        plantuml_set_makeprg = cfg.setMakeprg;
-        plantuml_executable_script = cfg.executableScript;
-      };
-    };
+  settingsOptions = {
+    set_makeprg = defaultNullOpts.mkBool true ''
```
Comment (traxys) [inline plugins/by-name/plantuml-syntax/default.nix:31]: Isn't this an `intFlag`? (@MattSturgeon)

Comment (khaneliman) [reply plugins/by-name/plantuml-syntax/default.nix:31]: Oh yeah... for vim plugins they use 0 or 1 right?

Comment (khaneliman) [reply plugins/by-name/plantuml-syntax/default.nix:31]: I thought we created a type/helper or something for that.

Comment (khaneliman) [reply plugins/by-name/plantuml-syntax/default.nix:31]: mkFlagInt ?

Comment (traxys) [reply plugins/by-name/plantuml-syntax/default.nix:31]: We have an `intFlag` type in our test lib, I don't remember how to use it

Comment (khaneliman) [reply plugins/by-name/plantuml-syntax/default.nix:31]: I see matt added `mkFlagInt` to our `defaultNullOpts` around this.

Comment (khaneliman) [inline plugins/by-name/plantuml-syntax/default.nix:31]:
```suggestion
    set_makeprg = defaultNullOpts.mkFlagInt 1 ''
```

**Review by khaneliman — APPROVED** [review]
Looks good now, thanks

**Comment by GaetanLepage** [comment]
@mergify queue

File: plugins/by-name/plantuml-syntax/default.nix (hunk 2 above)
Comment (GaetanLepage) [reply plugins/by-name/plantuml-syntax/default.nix]: Good catch. I probably missed a bunch in previous PRs.
I mean that I haven't fixed already "sub-optimal" option declarations.

Comment (khaneliman) [reply plugins/by-name/plantuml-syntax/default.nix]: That was all @traxys :P He just reminded me it was a thing

**Pull request merged**